inspectit-oce-core/src/main/java/rocks/inspectit/oce/core/utils/CommonUtils.java

```diff
@@ -32,4 +34,28 @@ public static boolean contentsEqual(Set<?> setA, Set<?> setB) {
         return !setA.stream().anyMatch(elem -> !setB.contains(elem));
     }
 
+    /**
+     * Tries to find the given type in the given classloader or one of its parents.
+     * Hereby, the given imported packages are taken into account.
+     * In addition, java.lang is assumed to be an implicit import.
+     *
+     * @param typename the name of the type, can also be fully qualified
+     * @param context  the classloader to search in
+     * @param packages the imported packages, e.g. "java.util", "javax.servlet"
+     * @return the Class if it was found, null otherwise
+     */
+    public static Class<?> locateTypeWithinImports(String typename, ClassLoader context, Collection<String> packages) {
```

Comment:
nice solution.. I guess you used `Stream.concat` to preserve order of trying:

1. FQN
2. something with declared import
3. Java class

I guess this is fine imo, can you just format a bit code in this method so that this is more readable:

```
return Stream.concat(
                Stream.concat(
                    Stream.of(""),
                    packages.stream().map(s -> s + ".")
                ),
                Stream.of("java.lang.")
            )
            .flatMap(prefix -> {
                try {
                    return Stream.of(Class.forName(prefix + typename, false, context));
                } catch (Exception e) {
                    return Stream.empty();
                }
            }).findFirst().orElse(null);
```